lapis/scheduler.py

```diff
@@ -26,9 +26,10 @@ class CondorJobScheduler(object):
     :return:
     """
 
-    def __init__(self, job_queue):
+    def __init__(self, job_queue, fileprovider):
```

Comment:
The scheduler should not have access to the `fileprovider`.

Please consider to have a caching logic component running besides the scheduler and just looking at the job queue. Based on this queue your component can change the requirements of the job to actually influence the matching of the scheduler. This would then be very similar to what is happening in real.

Reply:
The ``job_queue`` is created in the simulator and access is given to the scheduler. Further the scheduler provides direct access to this queue. So I currently do not see the point why do have plugins. I first considered having a component running alongside the scheduler. If we gain in performance by having plugins, we could also decide on this.

Reply:
> The scheduler should not have access to the `fileprovider`.

The alternative to passing the `fileprovider` via the scheduler is to modify the `make_drone` function. Due to the way this function is created with `partial()` step by step it has to be passed as arg. Therefore drones would have to have a fileprovider. I'm not sure whether this is a good idea.